OsmAnd/src/net/osmand/plus/GPXDatabase.java

```diff
@@ -258,9 +258,47 @@ SQLiteConnection openConnection(boolean readonly) {
 				onUpgrade(conn, version, DB_VERSION);
 			}
 		}
+		cleanUpDatabase(conn);
 		return conn;
 	}
 
+	private void cleanUpDatabase(SQLiteConnection db) {
+        db.execSQL("DELETE FROM " + GPX_TABLE_NAME +
```

Comment:
What purpose of this query?

Reply:
This query deletes all duplicates, where fileDir and filName are the same (that was possible because of the old getFileDir method, if we have the migration, this is not needed, I think)

Reply:
As the decision is made, that the migration is not needed, this just speeds up the cleanup at the first time, because there are tens of thousands of duplicate database entries, if one has hundreds of gpx files in sub-sub-folders of rec (I had about 250 files in sub-sub-folders and about 12000 database entries, all duplicates of this 250 files).